File: src/engine/transpile/src/test/resources/golden/SYK005/LK_メッセージ内容.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK005 のデータ部レコード LK-メッセージ内容 を逐語対訳した自動生成コード。 */
public final class LK_メッセージ内容 {

    /** 01 LK-メッセージ内容 総バイト長 80。 */
    public static final int LENGTH = 80;

    private final byte[] data;

    public LK_メッセージ内容() {
        this.data = new byte[80];
    }

    public LK_メッセージ内容(byte[] source) {
        this.data = Arrays.copyOf(source, 80);
    }

    public byte[] data() {
        return data;
    }

    // LK-メッセージ内容
    public String get_LK_メッセージ内容() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 80);
    }
    public void set_LK_メッセージ内容(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 80, value);
    }
}

File: src/engine/transpile/src/test/resources/golden/SYK005/LK_メッセージ内容.py
"""SYK005 のデータ部レコード LK-メッセージ内容 を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class LK_メッセージ内容:
    """01 LK-メッセージ内容 総バイト長 80。"""

    LENGTH = 80

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(80)
        else:
            self._data = bytearray(data)
            if len(self._data) < 80:
                self._data.extend(b"\x00" * (80 - len(self._data)))

    @property
    def data(self):
        return self._data

    # LK-メッセージ内容
    def get_LK_メッセージ内容(self):
        return decode_alphanumeric(self._data, 0, 80)
    def set_LK_メッセージ内容(self, value):
        encode_alphanumeric(self._data, 0, 80, value)
